pkg/controller/association/secret.go

```diff
@@ -130,11 +130,11 @@ func (r UnmanagedAssociationConnectionInfo) Request(path string, jsonPath string
 	if err != nil {
 		return "", err
 	}
+	defer resp.Body.Close()
```

Comment:
Good catch, this one may lead to a [memory|fd|connection] leak. FWIW I wrote a [quick and dirty integration test](https://gist.github.com/barkbay/f88ee6d6eb60cbd72403d5af8be9c623) that demonstrates the leak, after a few iterations we have some `too many open files` errors.